File: leetcode/java/209.minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int s, int[] nums) {
        if(nums == null || nums.length == 0) return 0;
        int sum = 0;
        int res = Integer.MAX_VALUE;
        int i = 0;
        int j = 0;
        while(i < nums.length){
            sum += nums[i];
            i++;
            while(sum >= s){
                res = Math.min(res, i - j);
                sum -= nums[j];
                j++;
            }
        }
        return res == Integer.MAX_VALUE?0:res;
    }
}

File: leetcode/golang/209.minimum-size-subarray-sum.go
func minSubArrayLen(target int, nums []int) int {
	res := len(nums) + 1
	sum := 0
	left := 0
	for i, v := range nums {
		sum += v
		for sum >= target {
			res = min(res, i-left+1)
			sum -= nums[left]
			left++
		}
	}
	if res == len(nums)+1 {
		return 0
	}
	return res
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}